### PR TITLE
Dashboard chart x-axis days formula

### PR DESCRIPTION
**Pull request by msbodetti**
### References Issues:
https://app.asana.com/0/1158142652422330/1158261716166746/f
### Description:
Requested to check the dots dates range:

- If the ticket sales dates are more than 30 days, x-axis should display by day
- If the ticket sales dates are more than 30 days, x-axis should display by week

### Release Screenshots / Video:
https://share.getcloudapp.com/nOumZdn2
### Environment Variables:
 * No change

### CODE REVIEW

File: src/components/pages/admin/events/dashboard/summary/charts/TicketSalesChart.js [hunk 1]
```diff
@@ -75,6 +84,23 @@ class SalesLine extends Component {
...
+		// Adjust the x-axis to display by days or weeks
+		resultSet.series().forEach(function(s, index) {
+			if(s.series.length >= 0){
+				const firstDate = moment(s.series[0].category);
+				const lastDate = moment(s.series[s.series.length - 1].category);
+				const countDays = Math.abs(
+					moment(firstDate, TIME_FORMAT_YYYY_MM_DD)
+						.startOf("day")
+						.diff(moment(lastDate, TIME_FORMAT_YYYY_MM_DD).startOf("day"), "days")
+				);
```
Comment (Krakaw) [inline src/components/pages/admin/events/dashboard/summary/charts/TicketSalesChart.js]: Could this not be simplified into:
```
const firstDate = moment(s.series[0].category).startOf("day");
const lastDate = moment(s.series[s.series.length - 1].category).startOf("day");
const countDays = Math.abs(firstDate.diff(lastDate, "days"));
```

File: src/components/pages/admin/events/dashboard/summary/charts/TicketSalesChart.js [hunk 2]
```diff
@@ -148,15 +174,13 @@ class SalesLine extends Component {
...
 							fontColor: "#2C3136",
-							fontSize: "12px"
-
-							// callback: (label, index, labels) => {
-							// 	const { value } = labels[0];
-							// 	return moment(value).format("MMM YYYY");
-							// }
+							fontSize: "12px",
+							callback: (label, index, labels) => {
+								return moment(label, "YYYY-MM-DD").format("DD MMM");
```
Comment (Krakaw) [inline src/components/pages/admin/events/dashboard/summary/charts/TicketSalesChart.js]: Re-use the `TIME_FORMAT_YYYY_MM_DD` constant here

File: src/components/pages/admin/events/dashboard/summary/charts/TicketSalesChart.js (hunk 1 above)
Comment (msbodetti) [reply src/components/pages/admin/events/dashboard/summary/charts/TicketSalesChart.js]: Brilliant, thank you

**Comment by Krakaw** [comment]
It's looking good, just have a look at https://develop.bigneon.com/admin/events/f7ada60d-bad6-4eb4-a496-fd149138ce3a/dashboard is that the expected behaviour?

**Comment by msbodetti** [comment]
Hi @Krakaw I'm not too sure regarding the 0 dots/points on the chart. We can confirm with Robin/Jessie regarding it and make an update to remove them?

**Comment by Krakaw** [comment]
@msbodetti when a day has 0 sales just don't show the dot but leave it there to be hovered.

**Comment by msbodetti** [comment]
Thanks @Krakaw I've added in a check to hide the dot for 0 values but it's still there and can be hovered over. Screencast https://share.getcloudapp.com/jkuKgx6e

**Pull request merged**